mesonbuild/backend/backends.py

```diff
@@ -1171,7 +1171,12 @@ def get_custom_target_sources(self, target):
             elif isinstance(i, build.GeneratedList):
                 fname = [os.path.join(self.get_target_private_dir(target), p) for p in i.get_outputs()]
             elif isinstance(i, build.ExtractedObjects):
-                fname = [os.path.join(self.get_target_private_dir(i.target), p) for p in i.get_outputs(self)]
+                outputs = i.get_outputs(self)
+                if self.name == 'xcode':
+                    # Xcode is stupid and requires a completely different naming scheme.
+                    fname = outputs
+                else:
+                    fname = [os.path.join(self.get_target_private_dir(i.target), p) for p in outputs]
```

Comment:
This really feels like a layering violation. Maybe we need a separate method that the xcode backend can override?